osfbot/exts/moderation/logs.py

```diff
@@ -0,0 +1,243 @@
+from datetime import datetime
+from typing import List, Optional
+
+from disnake import (
+    ButtonStyle,
+    Component,
+    Embed,
+    Member,
+    Message,
+    RawMessageDeleteEvent,
+    TextChannel,
+)
+from disnake.ext.commands import Cog
+from disnake.ui import Button
+from disnake.utils import format_dt
+from loguru import logger
+
+from ...bot import Bot
+from ...constants import Channels, Colors
+
+
+class ModerationLogs(Cog):
+    """Used to log infractions and important actions in the specified channel."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.log_channel: Optional[TextChannel] = None
+        super().__init__()
+
+    async def post_message(
+        self,
+        embed: Embed,
+        components: List[Component] = None,
+    ) -> Optional[Message]:
+        """Send the given message in the log channel."""
+        if not self.log_channel:
```

Comment:
IMO, it is better to do `if self.dm_log_channel is None` instead of `if not self.dm_log_channel` as the latter can be confused with boolean types.